### PR TITLE
feat(ModalFooterButtons): support disable buttons and remove secondary

### PR DESCRIPTION
**Pull request by omerg-monday**
#### Basic

- [ ] Used plop (`npm run plop`) to create a new component.
- [X] PR has description.
- [ ] New component is functional and uses Hooks.
- [X] Component is written in TypeScript.

#### Style

- [ ] Styles are added to `NewComponent.modules.scss` file inside the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
- [ ] Styles are defined using [monday-ui-style](https://github.com/mondaycom/monday-ui-react-core/tree/monorepo-style/packages/monday-ui-style) tokens
- [ ] Displays correctly in all the themes

#### Storybook

- [ ] Stories were added to `/src/components/NewComponent/__stories__/NewComponent.mdx` file.
- [ ] Stories include all flows of using the component.
- [ ] Stories implemented using [vibe-storybook-components](https://github.com/mondaycom/vibe-storybook-components) components and tokens.

#### Tests

- [X] Tests are compliant with [TESTING_README.md](/packages/core/TESTING_README.md) instructions.

#### Accessibility

- [ ] Component is accessible.
- [ ] Component is compatible with [Vibe Accessibility Guidelines](https://style.monday.com/?path=/docs/foundations-accessibility--docs)

### CODE REVIEW

**Review by YossiSaadi — COMMENTED** [review]
Seems cool by me 💪🏼 

@talkor I was wondering if we should allow passing all props for each of the button
wdyt? should we consult design about it? as it is abusable

File: packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/ModalFooterButtons.tsx [hunk 1]
```diff
@@ -9,7 +9,9 @@ import ModalFooter from "../ModalFooter";
 
 export interface ModalFooterButtonsProps extends VibeComponentProps {
   primaryButtonText: string;
-  secondaryButtonText: string;
+  secondaryButtonText?: string;
+  disablePrimaryButton?: boolean;
+  disableSecondaryButton?: boolean;
```
Comment (talkor) [inline packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/ModalFooterButtons.tsx:14]: We don't want to disable the secondary, just the primary, it's a UX consideration

**Review by talkor — CHANGES_REQUESTED** [review]
Looks good! Thank you for contributing! Let's just remove the `disableSecondaryButton`

**Comment by talkor** [comment]
> Seems cool by me 💪🏼
> 
> @talkor I was wondering if we should allow passing all props for each of the button wdyt? should we consult design about it? as it is abusable

Yes it was the thought at first but designers don't want to give this flexibility for now

File: packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/ModalFooterButtons.tsx (hunk 1 above)
Comment (omerg-monday) [reply packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/ModalFooterButtons.tsx]: done

File: packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/__tests__/modalFooterButtons-tests.jest.tsx [hunk 2]
```diff
@@ -42,5 +42,20 @@ describe("ModalFooterButtons", () => {
...
+      const modalFooterComponent = renderComponent({
+        primaryButtonText: PRIMARY_BUTTON_TEXT,
+        secondaryButtonText: SECONDARY_BUTTON_TEXT,
+        disablePrimaryButton: true,
+        onPrimaryButtonClick,
+        onSecondaryButtonClick
+      });
+      fireEvent.click(modalFooterComponent.getByText(PRIMARY_BUTTON_TEXT));
+      expect(onPrimaryButtonClick).not.toBeCalled();
+      expect(onSecondaryButtonClick).not.toBeCalled();
```
Comment (talkor) [inline packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/__tests__/modalFooterButtons-tests.jest.tsx:58]: This is redundant no?

**Review by talkor — APPROVED** [review]
Awesome thanks!

File: packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/__tests__/modalFooterButtons-tests.jest.tsx (hunk 2 above)
Comment (omerg-monday) [reply packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/__tests__/modalFooterButtons-tests.jest.tsx:58]: this is a test to validate that the button is indeed disabled

Comment (talkor) [reply packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/__tests__/modalFooterButtons-tests.jest.tsx:58]: I mean just this line, there's nothing going on with the secondary button in this test, no?

**Comment by talkor** [comment]
Closing #2013 
Closing #2012

**Pull request merged**